samza-core/src/main/scala/org/apache/samza/config/StreamConfig.scala

```diff
@@ -19,64 +19,187 @@
 
 package org.apache.samza.config
 
+import org.apache.samza.config.JobConfig.Config2Job
+import org.apache.samza.system.SystemStream
 import org.apache.samza.util.Logging
+
 import scala.collection.JavaConversions._
-import org.apache.samza.system.SystemStream
 
 object StreamConfig {
-  // stream config constants
+  // Samza configs for streams
+  val SAMZA_PROPERTY = "samza."
+  val SYSTEM =                  SAMZA_PROPERTY + "system"
+  val PHYSICAL_NAME =           SAMZA_PROPERTY + "physical.name"
+  val MSG_SERDE =               SAMZA_PROPERTY + "msg.serde"
+  val KEY_SERDE =               SAMZA_PROPERTY + "key.serde"
+  val CONSUMER_RESET_OFFSET =   SAMZA_PROPERTY + "reset.offset"
+  val CONSUMER_OFFSET_DEFAULT = SAMZA_PROPERTY + "offset.default"
+
+  val STREAMS_PREFIX = "streams."
+  val STREAM_ID_PREFIX = STREAMS_PREFIX + "%s."
+  val SYSTEM_FOR_STREAM_ID = STREAM_ID_PREFIX + SYSTEM
+  val PHYSICAL_NAME_FOR_STREAM_ID = STREAM_ID_PREFIX + PHYSICAL_NAME
+  val SAMZA_STREAM_PROPERTY_PREFIX = STREAM_ID_PREFIX + SAMZA_PROPERTY
+
   val STREAM_PREFIX = "systems.%s.streams.%s."
```

Comment:
nit: I prefer to move this to some inner class of OldStreamConfig s.t. we can mark it deprecated and delete those later.